src/WebJobs.Script/Binding/ExtensionLoader.cs

```diff
@@ -0,0 +1,224 @@
+                {
+                    continue;
+                }
+
+                if (IsExtensionLoaded(type))
+                {
+                    continue;
+                }
+
+                var extensionConfigProvider = (IExtensionConfigProvider)Activator.CreateInstance(type);
+                LoadExtension(extensionConfigProvider, locationHint);
+            }
+        }
+
+        private bool IsExtensionLoaded(Type type)
+        {
+            var registry = _config.HostConfig.GetService<IExtensionRegistry>();
+            var extensions = registry.GetExtensions<IExtensionConfigProvider>();
+            foreach (var extension in extensions)
+            {
+                var loadedExtentionType = extension.GetType();
+                if (loadedExtentionType == type)
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        private void LoadExtension(IExtensionConfigProvider extensionConfigProvider, string locationHint = null)
+        {
+            var extensionConfigProviderType = extensionConfigProvider.GetType();
+            string extensionName = extensionConfigProviderType.Name;
+
+            string msg = null;
+            if (!string.IsNullOrEmpty(locationHint))
+            {
+                msg = $"Loaded binding extension '{extensionName}' from '{locationHint}'";
+            }
+            else
+            {
+                msg = $"Loaded custom extension '{extensionName}'";
+            }
+
+            _traceWriter.Info(msg);
+            _startupLogger.LogInformation(msg);
+            _config.HostConfig.AddExtension(extensionConfigProvider);
+        }
+
+        public void LoadBuiltinExtensions(IEnumerable<string> bindingTypes)
+        {
+            foreach (var bindingType in bindingTypes)
+            {
+                string assemblyQualifiedTypeName;
+                if (_builtinBindingTypes.TryGetValue(bindingType, out assemblyQualifiedTypeName))
+                {
+                    Type typeExtension = Type.GetType(assemblyQualifiedTypeName);
+                    if (typeExtension == null)
+                    {
+                        string errorMsg = $"Can't find binding provider '{assemblyQualifiedTypeName}' for '{bindingType}'";
+                        _traceWriter.Error(errorMsg);
+                        _startupLogger?.LogError(errorMsg);
+                    }
+                    else
+                    {
+                        IExtensionConfigProvider extension = (IExtensionConfigProvider)Activator.CreateInstance(typeExtension);
+                        LoadExtension(extension);
+                    }
+                }
+            }
+        }
+
+        public static IEnumerable<string> DiscoverBindingTypes(IEnumerable<FunctionMetadata> functions)
+        {
+            if (FeatureFlags.IsEnabled(ScriptConstants.FeatureFlagsEnableDynamicExtensionLoading))
```

Comment:
This is the change that addresses the issue - we now allow dynamic/lazy loading to be turned on via a feature flag

Reply:
Is there enough visibility for `AzureWebJobsFeatureFlags` - might need some docs on this

Reply:
I added it to https://github.com/Azure/azure-webjobs-sdk-script/wiki/Configuration-Settings